runtime/math_functions.cpp

```diff
@@ -209,6 +219,49 @@ int64_t f$getrandmax() noexcept {
   return f$mt_getrandmax();
 }
 
+int64_t f$random_int(int64_t l, int64_t r) noexcept {
+  if (unlikely(l > r)) {
+    string errMsg{"Argument #1 ($min) must be less than or equal to argument #2 ($max)"};
+    THROW_EXCEPTION(make_throwable<C$ValueError>(string(__FILE__), __LINE__, 0, errMsg));
+    return {};
+  }
+
+  if (unlikely(l == r)) {
+    return l;
+  }
+
+  try {
+    std::random_device rd{"/dev/urandom"};
+    std::uniform_int_distribution dist{l, r};
+
+    return dist(rd);
+  } catch (const std::exception &e) {
+    string errMsg{"Source of randomness cannot be found"};
+    THROW_EXCEPTION(make_throwable<C$Random$RandomException>(string(__FILE__), __LINE__, 0, errMsg));
+    return -1;
+  } catch (...) {
+    php_critical_error("Unhandled exception");
+  }
+}
+
+string f$random_bytes(int64_t length) noexcept {
+  if (unlikely(length < 1)) {
+    string errMsg{"Argument #1 ($length) must be greater than 0"};
+    THROW_EXCEPTION(make_throwable<C$ValueError>(string(__FILE__), __LINE__, 0, errMsg));
+    return {};
+  }
+
+  string str{static_cast<string::size_type>(length), true};
```

Comment:
Actually, now it creates string with size=0, e.g. empty string for php developer